### PR TITLE
Adding response content type validation to the operation `getFileAsByteArray`

### PR DESCRIPTION
**Pull request by SanduDS**
# Description

Adding response content type validation to the operation `getFileAsByteArray`;
Adding support for specific byte range downloadings

Fixes https://github.com/ballerina-platform/ballerina-extended-library/issues/326

One line release note: 
- Adding response content type validation to the operation `getFileAsByteArray` and support for byte-range downloads

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
- All tests passed

**Test Configuration**:
* Ballerina Version: Swan Lake 2201.0.2
* Operating System:
* Java SDK: 11

# Checklist:

### Security checks
 - [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? 
 - [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: storageservice/modules/files/records.bal [hunk 1]
```diff
@@ -335,6 +335,15 @@ public type OptionalURIParametersFileMetaData record {|
     string timeout?;
 |};
 
+# Defines return bytes from the offset startByte through endByte
+#
+# + startByte - start position eg: 0  
+# + endByte - end position eng: 255
```
Comment (LakshanSS) [inline storageservice/modules/files/records.bal:341]: Check typo

File: storageservice/modules/files/file_client.bal [hunk 2]
```diff
@@ -376,71 +376,82 @@ public isolated client class FileClient {
...
-    remote isolated function getFileAsByteArray(@display {label: "File Share Name"} string fileShareName, 
-                                     @display {label: "Azure File Name"} string fileName,  
-                                     @display {label: "Azure Directory Path"} string? azureDirectoryPath = ()) 
-                                     returns @display {label: "Response"} byte[]|error {
-        string requestPath = azureDirectoryPath is () ? (SLASH + fileShareName + SLASH + fileName) : (SLASH 
-            + fileShareName + SLASH + azureDirectoryPath + SLASH + fileName);    
+    remote isolated function getFileAsByteArray(@display {label: "File Share Name"} string fileShareName,
+                                    @display {label: "Azure File Name"} string fileName,
+                                    @display {label: "Azure Directory Path"} string? azureDirectoryPath = (),
+                                    @display {label: "Specified Byte Range"} ContentRange? range = ())
```
Comment (LakshanSS) [inline storageservice/modules/files/file_client.bal:421]:
```suggestion
                                    @display {label: "Byte Range"} ContentRange? range = ())
```

File: storageservice/modules/files/records.bal [hunk 3]
```diff
@@ -335,6 +335,15 @@ public type OptionalURIParametersFileMetaData record {|
     string timeout?;
 |};
 
+# Defines return bytes from the offset startByte through endByte
```
Comment (LakshanSS) [inline storageservice/modules/files/records.bal:338]:
```suggestion
# Defines byte range of a file content
```

**Review by LakshanSS — APPROVED** [review]
Approved with minor suggestions

File: storageservice/modules/files/file_client.bal [hunk 4]
```diff
@@ -376,71 +376,82 @@ public isolated client class FileClient {
...
+                                    returns @display {label: "Response"} byte[]|error {
+        string requestPath = azureDirectoryPath is () ? (SLASH + fileShareName + SLASH + fileName) : (SLASH
+            + fileShareName + SLASH + azureDirectoryPath + SLASH + fileName);
         http:Request request = new;
+        if (range != ()) {
+            string rangeHeader = "bytes=" + range.startByte.toString() + "-" + range.endByte.toString();
```
Comment (sachinira) [inline storageservice/modules/files/file_client.bal:427]: Why we are getting `startByte` and `endByte` as int if we already know it needs to be converted to string?

File: storageservice/modules/files/file_client.bal [hunk 5]
```diff
@@ -376,71 +376,82 @@ public isolated client class FileClient {
...
-            requestPath = requestPath.concat(QUESTION_MARK, self.azureConfig.accessKeyOrSAS.substring(1)); 
+            requestPath = requestPath.concat(QUESTION_MARK, self.azureConfig.accessKeyOrSAS.substring(1));
         }
         map<string> headerMap = populateHeaderMapFromRequest(request);
         http:Response response = check self.httpClient->get(requestPath, headerMap);
-        byte[] payload = check response.getBinaryPayload();
-        return payload;
+        boolean isFileTypeAvailable = response.hasHeader(X_MS_TYPE);
+        if (isFileTypeAvailable == true) {
+            byte[] payload = check response.getBinaryPayload();
```
Comment (sachinira) [inline storageservice/modules/files/file_client.bal:449]: We can directly return here?

File: storageservice/modules/files/file_client.bal [hunk 6]
```diff
@@ -376,71 +376,82 @@ public isolated client class FileClient {
...
+            return payload;
+        } else {
+            xml errorMessage = check response.getXmlPayload();
+            fail error(errorMessage.toString());
```
Comment (sachinira) [inline storageservice/modules/files/file_client.bal:453]: What is the use of adding fail here?

File: storageservice/modules/files/file_client.bal (hunk 4 above)
Comment (LakshanSS) [reply storageservice/modules/files/file_client.bal:427]: if `startByte` and `endByte` are defined as strings, user can pass any other invalid inputs.
Keeping it as an `int` will give a good experience comparing to `string`

File: storageservice/modules/files/file_client.bal (hunk 5 above)
Comment (SanduDS) [reply storageservice/modules/files/file_client.bal:449]: This was the fix we added, Earlier we returned directly but for an incorrect file path, the error message also comes in the response body as XML file type with 200 status code. therefore, by using this workaround we can get rid of that issue

File: storageservice/modules/files/file_client.bal (hunk 6 above)
Comment (SanduDS) [reply storageservice/modules/files/file_client.bal:453]: This relates to the workaround, when there is an error message as XML payload, we provide it as an error to the user. please refer to the related issue for this bug fixing.

**Pull request merged**